zkvm/tests/zkvm.rs

```diff
@@ -0,0 +1,72 @@
+use bulletproofs::r1cs;
+use bulletproofs::{BulletproofGens, PedersenGens};
+use curve25519_dalek::ristretto::CompressedRistretto;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+use std::collections::VecDeque;
+
+use zkvm::*;
+
+fn issue_contract(
+    qty: u64,
+    flv: Scalar,
+    issuance_pred: &Predicate,
+    nonce_pred: &Predicate,
+) -> Vec<Instruction> {
+    vec![
+        Instruction::Push(Commitment::from(CommitmentWitness::blinded(qty)).into()), // stack: qty
+        Instruction::Var, // stack: qty-var
+        Instruction::Push(Commitment::from(CommitmentWitness::unblinded(flv)).into()), // stack: qty-var, flv
+        Instruction::Var,                                // stack: qty-var, flv-var
+        Instruction::Push(issuance_pred.clone().into()), // stack: qty-var, flv-var, pred
+        Instruction::Issue,                              // stack: issue-contract
+        Instruction::Push(nonce_pred.clone().into()),    // stack: issue-contract, pred
+        Instruction::Nonce,                              // stack: issue-contract, nonce-contract
+        Instruction::Signtx,                             // stack: issue-contract
+        Instruction::Signtx,                             // stack: issued-value
+        Instruction::Push(issuance_pred.clone().into()), // stack: issued-value, pred
```

Comment:
The issuance predicate is reused as an "account" for holding the issued token. That's totally fine, but to make the example more relevant to real usage, maybe add a third predicate: `recipient_pred`? To make it clear that we are sending funds somewhere.